producer/src/blockchain/blockchain_test.go

```diff
@@ -321,3 +321,61 @@ func TestRecoverBlockchainMetadata(t *testing.T) {
 		t.Errorf("Blocks do not match (block 2 by height)")
 	}
 }
+
+func TestGetYoungestBlockAndBlockHeader(t *testing.T) {
+	blockchain := "testBlockchain.dat"
+	table := "testTable.dat"
+	setUp(blockchain, table)
+	defer tearDown(blockchain, table)
+	_, err := GetYoungestBlock(blockchain, table)
+	if err == nil {
+		t.Errorf("Should return error if blockchain is empty")
+	}
+	block0 := block.Block{
+		Version:        1,
+		Height:         0,
+		Timestamp:      time.Now().UnixNano(),
+		PreviousHash:   block.HashSHA256([]byte{'0'}),
+		MerkleRootHash: block.HashSHA256([]byte{'1'}),
+		Data:           [][]byte{block.HashSHA256([]byte("xoxo"))},
+	}
+	block0.DataLen = uint16(len(block0.Data))
+	err = AddBlock(block0, blockchain, table)
+	if err != nil {
+		t.Errorf("Failed to add block")
+	}
+	actualBlock0, err := GetYoungestBlock(blockchain, table)
+	if err != nil {
+		t.Errorf("Error extracting youngest block")
+	}
+	if !cmp.Equal(actualBlock0, block0) {
+		t.Errorf("Blocks do not match")
+	}
+	block1 := block.Block{
+		Version:        1,
+		Height:         1,
+		Timestamp:      time.Now().UnixNano(),
+		PreviousHash:   block.HashSHA256([]byte{'0'}),
+		MerkleRootHash: block.HashSHA256([]byte{'1'}),
+		Data:           [][]byte{block.HashSHA256([]byte("xoxo"))},
+	}
+	block1.DataLen = uint16(len(block1.Data))
+	block1Header := block.BlockHeader{
+		Version:        1,
+		Height:         1,
+		Timestamp:      time.Now().UnixNano(),
```

Comment:
@HarryL5004  I think I figured out the issue. I suspected this might be a timestamp issue. The test is running faster on our machines than on Travis CI, so the timestamps are different between `block1` and `block1header`. Try changing this line to `Timestamp: block1.Timestamp`. That should fix it.

Reply:
good news, the build passed

Reply:
Okay, go ahead and make a new PR or update this branch with the appropriate changes. Choice is yours.